### ISSUE TITLE
Hello, I found that the pre trained MAE weights (PS) were directly used in your code. I'm not sure if I didn't find the pre trained code. If you haven't uploaded it yet, can you upload the pre trained model of MAE?

### ISSUE THREAD

**Issue by xiaochen217** [Henry-Lee-real/MaeFuse#6]
Hello, I found that the pre trained MAE weights (PS) were directly used in your code. I'm not sure if I didn't find the pre trained code. If you haven't uploaded it yet, can you upload the pre trained model of MAE?

**Comment by Henry-Lee-real** [comment]
We started from the official MAE large pretrained weights. Because the original MAE task differs from low-level vision, we reduced the number of decoder layers while keeping the encoder intact, and retrained only the decoder (a strategy later adopted by other papers as well). The resulting checkpoint is the ‘pre-trained initialization weights’ we provide in the README.”

**Comment by xiaochen217** [comment]
Where can I see the reduced number of layers, or your revised MAE, or where can I find related papers?

**Comment by Henry-Lee-real** [comment]
The file models_mae.py already contains the model-parameter settings (both the original and the modified versions).
The weights corresponding to the modified version are listed in the “How to use” section of the README under “Pre-train Weight for MAE.”

**Comment by xiaochen217** [comment]
I want to pre train the MAE model code, not just the weights, because I think your modified MAE works very well and I would like to refer to it







向恶势力低头
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Henry-Lee-real/MaeFuse"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年6月28日(星期六) 下午3:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Henry-Lee-real/MaeFuse] Hello, I found that the pre trained MAE weights (PS) were directly used in your code. I'm not sure if I didn't find the pre trained code. If you haven't uploaded it yet, can you upload the pre trained model of MAE? (Issue #6)



Henry-Lee-real left a comment (Henry-Lee-real/MaeFuse#6)
 
The file models_mae.py already contains the model-parameter settings (both the original and the modified versions).
 The weights corresponding to the modified version are listed in the “How to use” section of the README under “Pre-train Weight for MAE.”

**Comment by Henry-Lee-real** [comment]
My pretraining code is actually integrated into the existing codebase. In models_mae.py, I have already frozen the weights of the encoder. You only need to adjust the number of decoder layers through the configuration file. The loss function is also straightforward: it's simply the MSE loss between the decoder's output and the original image.

**Comment by wenchunchia** [comment]
> We started from the official MAE large pretrained weights. Because the original MAE task differs from low-level vision, we reduced the number of decoder layers while keeping the encoder intact, and retrained only the decoder (a strategy later adopted by other papers as well). The resulting checkpoint is the ‘pre-trained initialization weights’ we provide in the README.”

Thanks for the great work. I wonder where is the  ‘pre-trained initialization weights’ .

**Comment by Henry-Lee-real** [comment]
![Image](https://github.com/user-attachments/assets/0dac2a67-0d6a-48d3-96d2-bc6baefd0cf5)

**Comment by wenchunchia** [comment]
Could you please provide more details about the training process from the original MAE model to the pre-trained model that you have provided? I would like to start with the official MAE model. How should I select the model version and related configurations?

Thanks.

**Comment by Henry-Lee-real** [comment]
https://github.com/Henry-Lee-real/MaeFuse/blob/main/Pre-training-process.md I have update the guidance.

**Comment by wenchunchia** [comment]
> https://github.com/Henry-Lee-real/MaeFuse/blob/main/Pre-training-process.md I have update the guidance.

Thanks. And is the pre-trained input data also based on the MSRS dataset? If so, will the visible light y component and infrared images still be used? like the "MatchingImageDataset_GRAY" that you provided in the code.

**Comment by Henry-Lee-real** [comment]
Since the encoder is frozen (its representation capability is also fixed), we are only retraining the decoder to build reconstruction ability, so you can use any images you like (I used COCO).
